solana/solana-ibc/programs/solana-ibc/src/transfer/impls.rs

```diff
@@ -295,26 +295,131 @@ impl TokenTransferValidationContext for IbcStorage<'_, '_> {
 
     fn send_coins_validate(
         &self,
-        _from_account: &Self::AccountId,
-        _to_account: &Self::AccountId,
-        _coin: &PrefixedCoin,
+        from_account: &Self::AccountId,
+        to_account: &Self::AccountId,
+        coin: &PrefixedCoin,
     ) -> Result<(), TokenTransferError> {
-        Ok(())
+        /*
+           Should have the following accounts
+           - token program
+           - token account
+           - escrow account ( with seeds as portId, channelId and denom )
+           - token mint
+
+           If sending tokens from escrow then,
+           - mint authority should be present
+           - from account should match escrow
+           - to account should match token account
+
+          If sending tokens to escrow then,
+           - sender should be present
+           - sender should be signer
+           - to account should match escrow
+           - from account should match token account
+        */
+        let store = self.borrow();
+        let accounts = &store.accounts;
+        if accounts.token_program.is_none() ||
+            accounts.token_mint.is_none() ||
+            accounts.escrow_account.is_none() ||
+            accounts.token_account.is_none()
```

Comment:
The last two aren’t necessary now.
```suggestion
            accounts.token_mint.is_none()
```